eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -2487,4 +2490,34 @@ default void appendString(Appendable appendable, String separator)
         this.forEach(each -> target.put(groupByFunction.apply(each), collectFunction.apply(each)));
         return target;
     }
+
+    /**
+     * Converts the RichIterable to the default ImmutableList implementation.
+     *
+     * @since 11.0
+     */
+    default ImmutableList<T> toImmutableList()
+    {
+        return this.toList().toImmutable();
+    }
+
+    /**
+     * Converts the RichIterable to the default ImmutableSet implementation.
+     *
+     * @since 11.0
+     */
+    default ImmutableSet<T> toImmutableSet()
+    {
+        return this.toSet().toImmutable();
+    }
+
+    /**
+     * Converts the RichIterable to the default ImmutableBag implementation.
+     *
+     * @since 11.0
+     */
+    default ImmutableBag<T> toImmutableBag()
+    {
+        return this.toBag().toImmutable();
+    }
```

Comment:
These should live with the other to*() methods, probably right after toBiMap().

The other to*() methods are not default - why the difference?

Reply:
toImmutableSortedSet(), toImmutableSortedBag(), toImmutableMap(), toImmutableSortedMap()? Separate PR?

Reply:
Why prefer `this.to<type>().toImmutable()` over `<type>s.immutable.withAll(this)`?

Reply:
`to<type>()` implementations appear to have various overrides, so the code is not the same.

Reply:
@motlin Yes, I was planning for separate PRs for the other to methods since they have more overrides. I wanted to see if these made sense first.

Reply:
> Why prefer this.to<type>().toImmutable() over <type>s.immutable.withAll(this)?

My gut feel is that this made the most sense for the default case, but happy to change the default implementation. I'm also curious if there might be opportunities for optimizations in various implementations.

Reply:
Both have different opportunities for optimization. My thought is that if both are as optimized as possible, the `this.to<type>().toImmutable()` form will still make one more copy.

It kind of doesn't matter. Even if these methods are "slow" I'd still use them. It's just hard to see in a code review where these methods ought to be overridden imagining if they were not default methods.

Reply:
> Both have different opportunities for optimization. My thought is that if both are as optimized as possible, the `this.to<type>().toImmutable()` form will still make one more copy.
> 
> It kind of doesn't matter. Even if these methods are "slow" I'd still use them. It's just hard to see in a code review where these methods ought to be overridden imagining if they were not default methods.

For this PR I was really just looking for confirmation that these method names and signatures make sense. We can choose to optimize them by reducing copying. This will further illustrate their value and support the cost of adding them.

Reply:
@motlin @mohrezaei I'm going to change the code to use Craig's suggested approach. Another benefit is that all of the factory methods are optimized for empty.

Reply:
`withAll` creates an intermediate array. It would be best to get rid of that first. There is some optimization around small sets that appears to be buggy and should be fixed at the same time. Instead of creating an array, the iterable should be used to construct a UnifiedSet. If the set size is small, then it should be converted to a small immutable set, otherwise wrapped to immutable.
bug: pass in a list of size 1000 that has the same item. Instead of a singleton immutable set, the current code creates an wrapped unified set with a single entry.

Reply:
@mohrezaei These changes definitely make sense. I think we need to revisit the `withAll` implementations for the immutable factories across the board. I will create issues for these.